### PR TITLE
NCBI data source switch

### PR DESCRIPTION
**Pull request by Valiec**
This just reconfigures the summarize_variants pipeline to use NCBI data instead of GISAID, as well as a couple changes elsewhere resulting from that (specifically, the change in geographical precision, as well as using accession as an identifier instead of strain).

This was able to load data in my test last week, but there could still be some bugs I've missed.

### CODE REVIEW

File: lib/parse_ncbi.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            get_if_exists(line, "releaseDate"),
+            "" # sequence
+            ]
+
+# read seqs
+with open(sys.argv[3]) as f:
+    with open(sys.argv[4], "w") as g:
+        g.write("accession\tstrain\tdateCollected\tregion\tcountry\tdivision\tdateReleased\tsequence\n")
+
+        do_print = False
```
Comment (bwlang) [inline lib/parse_ncbi.py:61]: would it it be simpler to seqtk seq | paste -- to get a 2 column tsv and avoid having to keep state?
or maybe rename to "new_accession"?  or something else to communicate why you are printing

**Review by bwlang — APPROVED** [review]
couple of comments, but nothing very substantive.

File: lib/summarize_variants.nf [hunk 2]
```diff
@@ -39,28 +42,29 @@ process extract_new_records {
...
 
     shell:
     '''
     date_today=$(date +%Y-%m-%d)
 
-    python3 !{primer_monitor_path}/lib/filter_duplicates.py <(zstd -d --long=30 < !{prev_json}) <(zstd -d --long=30 < !{full_json}) > ${date_today}.json
+    python !{primer_monitor_path}/lib/parse_ncbi.py <(zstd -d --long=30 < !{metadata_json}) <(zstd -d --long=30 < !{prev_json}) <(zstd -d --long=30 < !{sequences_fasta}) ${date_today}.tsv
 
-    find !{output_path} -maxdepth 1 -mtime +5 -type f -name "*.full_json*"  -delete
+    find !{output_path} -maxdepth 1 -mtime +5 -type f -name "*.metadata.zst" -delete
```
[the remark this answers is not in the document]
Comment (Valiec) [reply lib/summarize_variants.nf:56]: Yes, I didn't know find could do more than one match in a command.

**Comment by Valiec** [comment]
I've made those fixes, but I still need to test them, don't merge yet.

**Comment by Valiec** [comment]
It seems to no longer crash, and the output looks valid.

File: lib/summarize_variants.nf (hunk 2 above)
[the remark this answers is not in the document]
Comment (Valiec) [reply lib/summarize_variants.nf]: Fixed.

File: lib/parse_ncbi.py (hunk 1 above)
Comment (Valiec) [reply lib/parse_ncbi.py]: Fixed, it now uses seqtk and paste to read in the seq data all on one line, so do_print is gone.

**Comment by Valiec** [comment]
The test has gotten far enough (at least one successful task for transform_data, align, and pangolin_calls) that I'd say it's working fine.

**Pull request merged**